File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count=0
        value=0
        for num in nums:
            if(count==0):
                value=num
            if(num==value):
                count+=1
            else:
                count-=1
        return value

        

        

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int freq=0,ans=0;
        int n=nums.size();
        for(int i=0;i<n;i++)
        {
            if(freq==0)ans=nums[i];
            if(ans==nums[i])freq++;
            else freq--;
        }
        int c=0;
        for(int val:nums)
        {
            if(val==ans)c++;
        }
        if(c>n/2)return ans;
        else return -1;
        //return ans;
        
        
    }
};